tests/unit/maxmemory.tcl

```diff
@@ -415,17 +415,33 @@ test_slave_buffers {slave buffer are counted correctly} 1000000 10 0 1
 test_slave_buffers "replica buffer don't induce eviction" 100000 100 1 0
 
 start_server {tags {"maxmemory external:skip"}} {
-    test {Don't rehash if used memory exceeds maxmemory after rehash} {
+    test {Don't resize if used memory exceeds maxmemory after resize} {
         r config set latency-tracking no
         r config set maxmemory 0
         r config set maxmemory-policy allkeys-random
 
         # Next rehash size is 8192, that will eat 64k memory
-        populate 4096 "" 1
+        populate 4095 "" 1
+
+        # Before adding a key to meet the 1:1 radio, disable resize to
+        # prevent the dict from being resized in cron.
+        r config set rdb-key-save-delay 10000000
+        r bgsave
+        r set k0 v0
```

Comment:
i don't understand why does cron triggers a resize in this case. isn't the size set by populate sufficient?
why do we need to add yet another key (both k0 and k1)?

Reply:
before the test call `populate 4096 "" 1`, it will set 4069 keys, this number of keys is enough for resize, 4096 is exactly 1:1. see tryResizeHashTables and dictExpandIfNeeded.

so I changed it to 4095 to avoid hit the radio, so in addition i add a k0 to remain the same as before.

we only checked 1:1 before add, so the test set a k1 key before, but k2 does seem to be unnecessary (I just don’t want to modify it)

Reply:
so the original test was wrong to use 4096, and should have used 4095 instead.
if it did that, then no rehashing would be triggered by cron until we add another key, and that's after changing the memory limit.
am i missing anything?

Reply:
> so the original test was wrong to use 4096, and should have used 4095 instead.

not quite right. previously, when the number of keys reached 4096, resize would not actually be triggered. We needed to perform an add operation before it would check and resize, like
```
set k v
before add key: check if we hit 1:1 radio, 4095 can not trigger the resize
add the key, now it is 4096

set k2 v2
before add key, check 1:1, 4096 hit the 1:1, so we will resize
add the key, not it is 4097
```

that is, before, we checked whether to resize before adding the key. After adding the key, although it reached 1:1, we will not be resized (if there is no write traffic)

and now in addition we will check in cron whether it reaches 1:1

Reply:
that's an off-by-one, (the check can probably check size+1 when done before adding).
but anyway, if we populate to 4095. and then add 2 keys (after changing the memory limit).
then why do we need the bgsave?

Reply:
yes, there is no need bgsave, i must be a litter off online.
i actually want to retain its previous testing intention (only check when adding keys)